Compilation problems

I try to compile each examples on ESP8266 but i receive this error. ESP32 similar problem.
Works fine with Arduino UNO/NANO/MEGA.
I am using IDE 1.8.14 or PlatformIO. 

```
surby\Documents\Arduino\libraries\MAX3010x_Sensor_Library\src/MAX30101.h:8,
                 from C:\Users\surby\Documents\Arduino\libraries\MAX3010x_Sensor_Library\src/MAX3010x.h:23,
                 from C:\Users\surby\Documents\Arduino\libraries\MAX3010x_Sensor_Library\examples\MAX30105PulseoximeterMultiLED\MAX30105PulseoximeterMultiLED.ino:1:
C:\Users\surby\Documents\Arduino\libraries\MAX3010x_Sensor_Library\src/MAX3010x_multiLed_core.h: In member function 'bool MAX3010xMultiLed<MAX3010xImpl, MAX3010xSample>::setMultiLedConfigurationInternal(uint8_t, uint8_t*)':
C:\Users\surby\Documents\Arduino\libraries\MAX3010x_Sensor_Library\src/MAX3010x_multiLed_core.h:85:12: error: there are no arguments to 'clearFIFO' that depend on a template parameter, so a declaration of 'clearFIFO' must be available [-fpermissive]
   85 |     return clearFIFO();

```
Any suggestion will be appreciatte!

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=MAX3010x Sensor Library
-version=1.0.1
+version=1.0.2
 author=Daniel Wiese
 maintainer=Daniel Wiese <contact@devxplained.eu>
 sentence=An Arduino library for the MAX3010x sensor family.

--- a/src/MAX30101.cpp
+++ b/src/MAX30101.cpp
@@ -52,7 +52,7 @@ MAX30101::MAX30101(uint8_t addr, TwoWire& wire) : MAX3010xMultiLed(addr, wire) {
  * @returns true if successful, otherwise false
  */
 bool MAX30101::setDefaultConfiguration() {
-  MultiLedConfiguration cfg = {0};
+  MultiLedConfiguration cfg {};
   
   if(!setMultiLedConfiguration(cfg)) return false;
   if(!setLedCurrent(LED_RED, 90)) return false;
@@ -96,7 +96,7 @@ bool MAX30101::setMultiLedConfiguration(const MAX30101::MultiLedConfiguration& c
     }
   }
   
-  uint8_t rawCfg[2] = {0};
+  uint8_t rawCfg[2] {};
   rawCfg[0] = static_cast<uint8_t>(cfg.slot[0]) | (static_cast<uint8_t>(cfg.slot[1]) << 4);
   rawCfg[1] = static_cast<uint8_t>(cfg.slot[2]) | (static_cast<uint8_t>(cfg.slot[3]) << 4);
   

--- a/src/MAX30102.cpp
+++ b/src/MAX30102.cpp
@@ -53,7 +53,7 @@ MAX30102::MAX30102(uint8_t addr, TwoWire& wire) : MAX3010xMultiLed(addr, wire) {
  * @returns true if successful, otherwise false
  */
 bool MAX30102::setDefaultConfiguration() {
-  MultiLedConfiguration cfg = {0};
+  MultiLedConfiguration cfg {};
   
   if(!setMultiLedConfiguration(cfg)) return false;
   if(!setLedCurrent(LED_RED, 90)) return false;
@@ -95,7 +95,7 @@ bool MAX30102::setMultiLedConfiguration(const MAX30102::MultiLedConfiguration& c
     }
   }
   
-  uint8_t rawCfg[2] = {0};
+  uint8_t rawCfg[2] {};
   rawCfg[0] = static_cast<uint8_t>(cfg.slot[0]) | (static_cast<uint8_t>(cfg.slot[1]) << 4);
   rawCfg[1] = static_cast<uint8_t>(cfg.slot[2]) | (static_cast<uint8_t>(cfg.slot[3]) << 4);
   

--- a/src/MAX30105.cpp
+++ b/src/MAX30105.cpp
@@ -56,7 +56,7 @@ MAX30105::MAX30105(uint8_t addr, TwoWire& wire) : MAX3010xMultiLed(addr, wire) {
  * @returns true if successful, otherwise false
  */
 bool MAX30105::setDefaultConfiguration() {
-  MultiLedConfiguration cfg = {0};
+  MultiLedConfiguration cfg {};
   
   if(!setMultiLedConfiguration(cfg)) return false;
   if(!setLedCurrent(LED_RED, 90)) return false;
@@ -119,7 +119,7 @@ bool MAX30105::setMultiLedConfiguration(const MAX30105::MultiLedConfiguration& c
     }
   }
   
-  uint8_t rawCfg[2] = {0};
+  uint8_t rawCfg[2] {};
   rawCfg[0] = static_cast<uint8_t>(cfg.slot[0]) | (static_cast<uint8_t>(cfg.slot[1]) << 4);
   rawCfg[1] = static_cast<uint8_t>(cfg.slot[2]) | (static_cast<uint8_t>(cfg.slot[3]) << 4);
   

--- a/src/MAX3010x_multiLed_core.h
+++ b/src/MAX3010x_multiLed_core.h
@@ -77,12 +77,12 @@ protected:
   bool setMultiLedConfigurationInternal(uint8_t activeSlots, uint8_t cfg[2]) {
     uint8_t value;
     
-    if(!writeBlock(MAX3010xImpl::MULTI_LED_CFG_REG_BASE, sizeof(cfg), cfg)) return false;
+    if(!MAX3010x<MAX3010xImpl, MAX3010xSample>::writeBlock(MAX3010xImpl::MULTI_LED_CFG_REG_BASE, sizeof(cfg), cfg)) return false;
     
     nConfiguredSlots = activeSlots;
     if(currentMode == MODE_MULTI_LED) nActiveSlots = nConfiguredSlots;
     
-    return clearFIFO();
+    return MAX3010x<MAX3010xImpl, MAX3010xSample>::clearFIFO();
   }
   
   /**
@@ -110,7 +110,7 @@ public:
     else if(mode == MODE_MULTI_LED) activeSlots = nConfiguredSlots;
     else return false;
     
-    if(!setModeInternal(static_cast<uint8_t>(mode))) return false;
+    if(!MAX3010x<MAX3010xImpl, MAX3010xSample>::setModeInternal(static_cast<uint8_t>(mode))) return false;
     
     currentMode = mode;
     nActiveSlots = activeSlots;
@@ -140,12 +140,12 @@ public:
     uint8_t cfg;
     
     if(rate & (~ SPO2_CFG_SMP_RATE_MASK)) return false;
-    if(!readByte(SPO2_CFG_REG, cfg)) return false;
+    if(!MAX3010x<MAX3010xImpl, MAX3010xSample>::readByte(SPO2_CFG_REG, cfg)) return false;
     
     cfg &= ~(SPO2_CFG_SMP_RATE_MASK << SPO2_CFG_SMP_RATE_BIT);
     cfg |= (static_cast<uint8_t>(rate) & SPO2_CFG_SMP_RATE_MASK) << SPO2_CFG_SMP_RATE_BIT;
     
-    return writeByte(SPO2_CFG_REG, cfg);
+    return MAX3010x<MAX3010xImpl, MAX3010xSample>::writeByte(SPO2_CFG_REG, cfg);
   }
   
   /**
@@ -167,12 +167,12 @@ public:
     uint8_t cfg;
     
     if(range & (~ SPO2_CFG_ADC_RANGE_MASK)) return false;
-    if(!readByte(SPO2_CFG_REG, cfg)) return false;
+    if(!MAX3010x<MAX3010xImpl, MAX3010xSample>::readByte(SPO2_CFG_REG, cfg)) return false;
     
     cfg &= ~(SPO2_CFG_ADC_RANGE_MASK << SPO2_CFG_ADC_RANGE_BIT);
     cfg |= (static_cast<uint8_t>(range) & SPO2_CFG_ADC_RANGE_MASK) << SPO2_CFG_ADC_RANGE_BIT;
     
-    return writeByte(SPO2_CFG_REG, cfg);
+    return MAX3010x<MAX3010xImpl, MAX3010xSample>::writeByte(SPO2_CFG_REG, cfg);
   }
   
   /**
@@ -194,12 +194,12 @@ public:
     uint8_t cfg;
     
     if(resolution & (~ SPO2_CFG_RESOLUTION_MASK)) return false;
-    if(!readByte(SPO2_CFG_REG, cfg)) return false;
+    if(!MAX3010x<MAX3010xImpl, MAX3010xSample>::readByte(SPO2_CFG_REG, cfg)) return false;
     
     cfg &= ~(SPO2_CFG_RESOLUTION_MASK << SPO2_CFG_RESOLUTION_BIT);
     cfg |= (static_cast<uint8_t>(resolution) & SPO2_CFG_RESOLUTION_MASK) << SPO2_CFG_RESOLUTION_BIT;
     
-    return writeByte(SPO2_CFG_REG, cfg);
+    return MAX3010x<MAX3010xImpl, MAX3010xSample>::writeByte(SPO2_CFG_REG, cfg);
   }
   
   /**
@@ -207,7 +207,7 @@ public:
   * @return true if successful, otherwise false
   */
   bool enableFIFORollover() {
-    return setBit(FIFO_CFG_REG, FIFO_ROLLOVER_EN_BIT, true);
+    return MAX3010x<MAX3010xImpl, MAX3010xSample>::setBit(FIFO_CFG_REG, FIFO_ROLLOVER_EN_BIT, true);
   }
   
   /**
@@ -215,7 +215,7 @@ public:
   * @return true if successful, otherwise false
   */
   bool disableFIFORollover() {
-    return setBit(FIFO_CFG_REG, FIFO_ROLLOVER_EN_BIT, false);
+    return MAX3010x<MAX3010xImpl, MAX3010xSample>::setBit(FIFO_CFG_REG, FIFO_ROLLOVER_EN_BIT, false);
   }
   
   /**
@@ -240,12 +240,12 @@ public:
     uint8_t cfg;
     
     if(averaging & (~ FIFO_SMP_AVE_MASK)) return false;
-    if(!readByte(FIFO_CFG_REG, cfg)) return false;
+    if(!MAX3010x<MAX3010xImpl, MAX3010xSample>::readByte(FIFO_CFG_REG, cfg)) return false;
     
     cfg &= ~(FIFO_SMP_AVE_MASK << FIFO_SMP_AVE_BIT);
     cfg |= (static_cast<uint8_t>(averaging) & FIFO_SMP_AVE_MASK) << FIFO_SMP_AVE_BIT;
     
-    return writeByte(FIFO_CFG_REG, cfg);
+    return MAX3010x<MAX3010xImpl, MAX3010xSample>::writeByte(FIFO_CFG_REG, cfg);
   }
   
 };

